packages/purpose-process/test/api/getRiskAnalysisDocument.test.ts

```diff
@@ -0,0 +1,106 @@
+  purposeVersionNotFound,
+} from "../../src/model/domain/errors.js";
+
+describe("API GET /purposes/{purposeId}/versions/{versionId}/documents/{documentId} test", () => {
+  const mockDocument = getMockPurposeVersionDocument();
+
+  const apiResponse = purposeApi.PurposeVersionDocument.parse(
+    purposeVersionDocumentToApiPurposeVersionDocument(mockDocument)
+  );
+
+  beforeEach(() => {
+    purposeService.getRiskAnalysisDocument = vi
+      .fn()
+      .mockResolvedValue(mockDocument);
+  });
+
+  const makeRequest = async (
+    token: string,
+    purposeId: string = generateId(),
+    versionId: string = generateId(),
+    documentId: string = generateId()
+  ) =>
+    request(api)
+      .get(
+        `/purposes/${purposeId}/versions/${versionId}/documents/${documentId}`
+      )
+      .set("Authorization", `Bearer ${token}`)
+      .set("X-Correlation-Id", generateId());
+
+  const authorizedRoles: AuthRole[] = [
+    authRole.ADMIN_ROLE,
+    authRole.SUPPORT_ROLE,
+  ];
+
+  it.each(authorizedRoles)(
+    "Should return 200 for user with role %s",
+    async (role) => {
+      const token = generateToken(role);
+      const res = await makeRequest(token);
+      expect(res.status).toBe(200);
+      expect(res.body).toEqual(apiResponse);
+    }
+  );
+
+  it.each(
+    Object.values(authRole).filter((role) => !authorizedRoles.includes(role))
+  )("Should return 403 for user with role %s", async (role) => {
+    const token = generateToken(role);
+    const res = await makeRequest(token);
+    expect(res.status).toBe(403);
+  });
+
+  it("Should return 404 for purposeNotFound", async () => {
+    purposeService.getRiskAnalysisDocument = vi
+      .fn()
+      .mockRejectedValue(purposeNotFound(generateId()));
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token);
+    expect(res.status).toBe(404);
+  });
+
+  it("Should return 404 for purposeVersionNotFound", async () => {
+    purposeService.getRiskAnalysisDocument = vi
+      .fn()
+      .mockRejectedValue(purposeVersionNotFound(generateId(), generateId()));
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token);
+    expect(res.status).toBe(404);
+  });
+
+  it("Should return 404 for purposeVersionDocumentNotFound", async () => {
+    purposeService.getRiskAnalysisDocument = vi
+      .fn()
+      .mockRejectedValue(
+        purposeVersionDocumentNotFound(generateId(), generateId(), generateId())
+      );
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token);
+    expect(res.status).toBe(404);
+  });
+
+  it("Should return 403 for organizationNotAllowed", async () => {
+    purposeService.getRiskAnalysisDocument = vi
+      .fn()
+      .mockRejectedValue(organizationNotAllowed(generateId()));
+    const token = generateToken(authRole.ADMIN_ROLE);
+    const res = await makeRequest(token);
+    expect(res.status).toBe(403);
+  });
+});
```

Comment:
I noticed that the OpenAPI spec for this endpoint doesn't mark `format: uuid` for the parameters, though I think at least `purposeId` and `versionId` should be UUIDs

This also means there's no 400 test because this endpoint doesn't return 400 if those parameters are formatted inconsistently or even empty

Should we add `format: uuid` to the spec in this PR? If so, is `documentId` also an UUID?

https://github.com/pagopa/interop-be-monorepo/blob/0a8263b8385cb0565f94f05c6643c461b1e846d8/packages/api-clients/open-api/purposeApi.yml#L540

Reply:
We discussed and decided to fix such issues in the supertest PRs directly, when they're minor: so I added the fix in 8341ea557793a554484472fdde7886e26887f871

I marked `documentId` as UUID as well as the other two since I saw it's defined as UUID in the BFF [link](https://github.com/pagopa/interop-be-monorepo/blob/8341ea557793a554484472fdde7886e26887f871/packages/models/src/brandedIds.ts#L65)